src/py/flwr/cli/ls.py

```diff
@@ -255,19 +257,66 @@ def _to_table(run_list: list[tuple[str, ...]]) -> Table:
     return table
 
 
+def _to_json(run_list: list[tuple[str, ...]]) -> str:
+    """Format run status list to a JSON formatted string."""
+
+    def _remove_bbcode_tags(strings: tuple[str, ...]) -> tuple[str, ...]:
+        """Remove BBCode tags from the provided text."""
+        # Regular expression pattern to match BBCode tags
+        bbcode_pattern = re.compile(r"\[/?\w+\]")
+        # Substitute BBCode tags with an empty string
+        return tuple(bbcode_pattern.sub("", s) for s in strings)
+
+    runs_list = []
+    for row in run_list:
+        row = _remove_bbcode_tags(row)
+        (
+            run_id,
+            fab_id,
+            fab_version,
+            fab_hash,
+            status_text,
+            elapsed,
+            created_at,
+            running_at,
+            finished_at,
+        ) = row
+        runs_list.append(
+            {
+                "run-id": run_id,
+                "fab-id": fab_id,
+                "fab-name": fab_id.split("/")[-1],
+                "fab-version": fab_version,
+                "fab-hash": fab_hash[:8],
+                "status": status_text,
+                "elapsed": elapsed,
+                "created-at": created_at,
+                "running-at": running_at,
+                "finished-at": finished_at,
+            }
+        )
+
+    return json.dumps({"runs": runs_list})
+
+
 def _list_runs(
     stub: ExecStub,
+    output_format: str = CliOutputFormat.default,
 ) -> None:
     """List all runs."""
     res: ListRunsResponse = stub.ListRuns(ListRunsRequest())
     run_dict = {run_id: run_from_proto(proto) for run_id, proto in res.run_dict.items()}
 
-    Console().print(_to_table(_format_runs(run_dict, res.now)))
+    if output_format == CliOutputFormat.json:
+        Console().print_json(_to_json(_format_runs(run_dict, res.now)))
```

Comment:
Oh, this is interesting. So the rich Console also supports printing json? What can this `print_json` do?

Reply:
Yes! (I thought I replied to this comment somewhere else). I didn't explore it further, but so far it produces nicely formatted JSON output.